earth_enterprise/src/common/gedbroot/gedbroot_google_url_remover.cpp

```diff
@@ -97,26 +97,32 @@ int main(int argc, char* argv[]) {
   if (cleaned_output_fname.empty()) {
     cleaned_output_fname = fname_dbroot + ".cleaned";
   }
-  geProtoDbroot dbroot(fname_dbroot, geProtoDbroot::kEncodedFormat);
-
-  bool dbroot_is_valid = dbroot.IsInitialized();
-  if (!dbroot_is_valid) {
-    notify(NFY_FATAL,
-           "Given proto dbroot is invalid; refusing to touch it further");
-    // Diagnose
-    dbroot.CheckInitialized();
-  } else {
-    if (want_debug) {
-      dbroot.Write(fname_dbroot + ".before.txt", geProtoDbroot::kTextFormat);
-    }
-
-    dbroot_is_valid = RemoveUnwantedGoogleUrls(&dbroot);
-    dbroot.Write(cleaned_output_fname, geProtoDbroot::kEncodedFormat);
-
-    if (want_debug) {
-      dbroot.Write(cleaned_output_fname + ".txt", geProtoDbroot::kTextFormat);
+  bool dbroot_is_valid = false;
+  try {
```

Comment:
Was there an uncaught exception that necessitated a try/catch? Does this end up calling `EtaDbroot::DecodeBinary` where the "new" throws are?

Reply:
This calls all the way down to Encode, but I think it goes through proto_dbroot.cc, not eta_dbroot.cc. There was already the possibility that an exception would be thrown and it wasn't catching it. I figured I'd fix it since I added another exception.